### DISCUSSION TITLE
How to use Vuex + Component Testing

### DISCUSSION THREAD

**Discussion by MaYoRr** [cypress-io/cypress#26751] [Component Testing]
Im using Cypress + Vue Js and Component testing. My component using Vuex. How to make test with vuex imitation. Now I have this error:
Cannot read properties of undefined (reading 'getters')

**Comment by sirSayed98** [comment]
you can use this snippet

create mock-vuex-store.js

`
import { createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';


export function storeMock(storeConfig, config) {
  const localVue = createLocalVue();
  localVue.use(Vuex);

  let store = new Vuex.Store(storeConfig);
  Object.assign(store, config);

  return { store, localVue };
}
`
in your test file

import { storeMock } from "your_path/mock-vuex-store";


`cy.mount(your_component_name, {
    ...storeMock({
      getters: {
      },
      modules: {
        your_module: {
          namespaced: true,
          state: {
          }
        }
      }
    })
  });
  `
